src/components/App.tsx

```diff
@@ -1,15 +1,30 @@
+import { AxiosError } from "axios";
 import * as React from "react";
 import { RouteProps } from "react-router-dom";
 import LiftLogService from "./../services/LiftLogService";
 import { LiftLogEntry } from "./../types/LiftTypes";
 import "./App.css";
 import LiftLogContainer from "./LiftLogContainer";
 
-type State = {
+type LoadingState = {
+  isLoading: true;
+};
+
+type ErrorState = {
+  isLoading: false;
+  networkErrorOccured: true;
```

Comment:
seems redundant to have both `networkErrorOccured` and `errorMessage` since you can have `errorMessage` as `null` which would indicate the absence of error. How does that sound? And btw, the correct spelling is `occuRRed` ;)

Reply:
The point of using an ADT is precisely that - avoiding nulls :)
I agree that it's redundant, but that's somewhat coincidental, since we have only one property - errorMessage. If we had e.g. `errorCode/errorMessage/somethingElse` I would be trickier to achieve the same effect without using a union type.
I will change the spelling, thanks for pointing that out! ;)